### PR TITLE
Add OPTIONS-method as a shortcut

### PR DESCRIPTION
**Pull request by theikkila**
Axios is missing OPTIONS-method as a shortcut. This PR will implement it

### CODE REVIEW

**Comment by nickuraltsev** [comment]
Thank you for the PR! What's the use case for this?

**Comment by theikkila** [comment]
This is for making the API consistent across all standardized HTTP-verbs. Options is sematically used ie. when fetching dynamic options for some endpoint.
Usually the response can tell the allowed or required fields.

**Comment by ianks** [comment]
I would like to use this to test that my API is returning the correct CORS headers.

**Comment by nickuraltsev** [comment]
I don't think we need to add a shortcut method for `OPTIONS` as preflight requests are normally sent by the browser. For test purposes, you can use the `request` method as follows:

```
axios.request({
  method: 'options',
 url: '/foo'
})
```

Thank you!

**Comment by theikkila** [comment]
`OPTIONS` are not just related to preflight requests but also for actually inspecting meta-data about some REST api endpoint. OPTIONS-shortcut is not for preflight requests but as I said  ie. to gather some options about the validation.

According to RFC-2616 / 9.2 OPTIONS:
https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html

> The OPTIONS method represents a request for information about the communication options available on the request/response chain identified by the Request-URI. This method allows the client to determine the options and/or requirements associated with a resource, or the capabilities of a server, without implying a resource action or initiating a resource retrieval.


Real world example is Django Rest Framework, it uses the OPTIONS-verb exacly for that purpose.

Please reopen this PR.

File: README.md [hunk 1]
```diff
@@ -142,6 +142,7 @@ For convenience aliases have been provided for all supported request methods.
 ##### axios.post(url[, data[, config]])
 ##### axios.put(url[, data[, config]])
 ##### axios.patch(url[, data[, config]])
+##### axios.options(url[, data[, config]])
```
Comment (nickuraltsev) [inline README.md]: The `options` method does not accept `data` parameter. Could you please remove `data` and place `options` between `head` and `post` in `README.md`?

Comment (theikkila) [inline README.md]: Yeah, sure!

**Comment by theikkila** [comment]
Fixed reviewed problems. Also updated the documentation from instance methods to contain the new added alias.

**Pull request merged**